app/src/main/java/com/duckduckgo/app/browser/duckchat/DuckChatPreferencesStore.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.duckchat
+
+import androidx.core.content.edit
+import com.duckduckgo.data.store.api.SharedPreferencesProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface DuckChatPreferencesStore {
+    fun fetchAndClearUserPreferences(): String?
+    fun updateUserPreferences(userPreferences: String?)
+}
+
+@ContributesBinding(AppScope::class)
+class RealDuckChatPreferencesStore @Inject constructor(
```

Comment:
nit: use DataStore instead? AFAIK we want to move away from shared preferences

Reply:
Will update in a follow-up